samples/nrf9160/coap_client/prj.conf

```diff
@@ -0,0 +1,29 @@
+# General config
+CONFIG_TEST_RANDOM_GENERATOR=y
+CONFIG_HEAP_MEM_POOL_SIZE=16384
+
+# Networking
+CONFIG_NETWORKING=y
+CONFIG_NET_SOCKETS_OFFLOAD=y
+CONFIG_NET_SOCKETS=y
+CONFIG_NET_SOCKETS_POSIX_NAMES=y
+
+# LTE link control
+CONFIG_LTE_LINK_CONTROL=y
+CONFIG_LTE_AUTO_INIT_AND_CONNECT=n
```

Comment:
This can be dropped, along with `modem_configure` in `main.c` as there's noe harm in setting up the link before application starts when certificates are not used. The downside is that users will not get the printk that the link is being set up. Considering that, it's possibly best to leave it as is.

Reply:
We do repeat this piece of code in every single sample. Coudn't we instead let the Link Control module to report it's status through logger instead? With this approach, we could remove the manual modem initialization from samples, while still getting the information needed.

The truth is that modem configuration is the only LTE-specifc piece of code in this sample and in MQTT sample as well. Getting rid of these would make the samples fully technology agnostic.

Reply:
Agreed, it would be nice to have these samples free of LTE specific code. The modem initialization has to stay in samples that can provision certificates, such as the asset tracker, as that has to happen before the link is set up. But for for CoAP and MQTT there's really no reason to keep it.